actors/abi/sector.go

```diff
@@ -69,75 +69,98 @@ type RegisteredProof int64
 	}
+}
 
+// RegisteredWinningPoStProof produces the PoSt-specific RegisteredProof corresponding
+// to the receiving RegisteredProof.
+func (p RegisteredProof) RegisteredWinningPoStProof() (RegisteredProof, error) {
+	switch p {
+	case RegisteredProof_StackedDRG32GiBSeal, RegisteredProof_StackedDRG32GiBWinningPoSt:
+		return RegisteredProof_StackedDRG32GiBWinningPoSt, nil
+	case RegisteredProof_StackedDRG2KiBSeal, RegisteredProof_StackedDRG2KiBWinningPoSt:
+		return RegisteredProof_StackedDRG2KiBWinningPoSt, nil
+	case RegisteredProof_StackedDRG8MiBSeal, RegisteredProof_StackedDRG8MiBWinningPoSt:
+		return RegisteredProof_StackedDRG8MiBWinningPoSt, nil
+	case RegisteredProof_StackedDRG512MiBSeal, RegisteredProof_StackedDRG512MiBWinningPoSt:
+		return RegisteredProof_StackedDRG512MiBWinningPoSt, nil
+	default:
+		return 0, errors.Errorf("unsupported mapping from %+v to PoSt-specific RegisteredProof", p)
+	}
 }
 
-// RegisteredPoStProof produces the PoSt-specific RegisteredProof corresponding
+// RegisteredWindowPoStProof produces the PoSt-specific RegisteredProof corresponding
 // to the receiving RegisteredProof.
-func (p RegisteredProof) RegisteredPoStProof() (RegisteredProof, error) {
+func (p RegisteredProof) RegisteredWindowPoStProof() (RegisteredProof, error) {
 	switch p {
-	case RegisteredProof_StackedDRG32GiBSeal:
-		return RegisteredProof_StackedDRG32GiBPoSt, nil
-	case RegisteredProof_StackedDRG32GiBPoSt:
-		return RegisteredProof_StackedDRG32GiBPoSt, nil
-	case RegisteredProof_StackedDRG2KiBSeal:
-		return RegisteredProof_StackedDRG2KiBPoSt, nil
-	case RegisteredProof_StackedDRG2KiBPoSt:
-		return RegisteredProof_StackedDRG2KiBPoSt, nil
-	case RegisteredProof_StackedDRG8MiBSeal:
-		return RegisteredProof_StackedDRG8MiBPoSt, nil
-	case RegisteredProof_StackedDRG8MiBPoSt:
-		return RegisteredProof_StackedDRG8MiBPoSt, nil
-	case RegisteredProof_StackedDRG512MiBSeal:
-		return RegisteredProof_StackedDRG512MiBPoSt, nil
-	case RegisteredProof_StackedDRG512MiBPoSt:
-		return RegisteredProof_StackedDRG512MiBPoSt, nil
+	case RegisteredProof_StackedDRG32GiBSeal, RegisteredProof_StackedDRG32GiBWindowPoSt:
+		return RegisteredProof_StackedDRG32GiBWindowPoSt, nil
+	case RegisteredProof_StackedDRG2KiBSeal, RegisteredProof_StackedDRG2KiBWindowPoSt:
+		return RegisteredProof_StackedDRG2KiBWindowPoSt, nil
+	case RegisteredProof_StackedDRG8MiBSeal, RegisteredProof_StackedDRG8MiBWindowPoSt:
+		return RegisteredProof_StackedDRG8MiBWindowPoSt, nil
+	case RegisteredProof_StackedDRG512MiBSeal, RegisteredProof_StackedDRG512MiBWindowPoSt:
+		return RegisteredProof_StackedDRG512MiBWindowPoSt, nil
 	default:
 		return 0, errors.Errorf("unsupported mapping from %+v to PoSt-specific RegisteredProof", p)
 	}
 }
 
+// RegisteredPoStProof produces the PoSt-specific RegisteredProof corresponding
+// to the receiving RegisteredProof.
+func (p RegisteredProof) RegisteredPoStProof() (RegisteredProof, error) {
+	panic("unsupported method")
```

Comment:
let's not have a panic here. either delete the method if we don't want it called, or return an error